File: scripts/parsers/parse_all_admision.py
import re
import json
import os

with open("Raz verbal/preguntas_admision.md", "r", encoding="utf-8") as f:
    content = f.read()

questions = []
current_area = "Razonamiento Verbal"
current_tema = "General"
current_tipo = "opcion_multiple"
current_context = ""

# Parse sections and questions
lines = content.split('\n')
i = 0
while i < len(lines):
    line = lines[i].strip()
    
    if line.startswith("### EXAMEN: DPTO. DE CIENCIAS"):
        current_area = "Razonamiento Lógico"
        current_tema = "Resolución de Problemas"
    elif line.startswith("#### SECCIÓN:"):
        tema_str = line.replace("#### SECCIÓN:", "").strip()
        if "ORTOGRAFÍA" in tema_str or "ACENTUACIÓN" in tema_str:
            current_tema = "Corrección Ortográfica"
            current_tipo = "comparacion_ortografica"
        elif "SINÓNIMOS" in tema_str:
            current_tema = "Sinónimos en Contexto"
            current_tipo = "opcion_multiple"
        elif "INCOMPLETAS" in tema_str or "COMPLETACIÓN" in tema_str:
            current_tema = "Completación de Oraciones"
            current_tipo = "opcion_multiple"
        elif "ANALOGÍAS" in tema_str:
            current_tema = "Analogías y Relaciones Semánticas"
            current_tipo = "opcion_multiple"
        elif "RAZONAMIENTO LÓGICO" in tema_str:
            current_tema = "Lógica Deductiva"
            current_tipo = "opcion_multiple"
        elif "COMPRENSIÓN DE LECTURA" in tema_str:
            current_tema = "Comprensión Lectora"
            current_tipo = "opcion_multiple"
        elif "ORDENACIÓN" in tema_str:
            current_tema = "Ordenación Lógica de Párrafos"
            current_tipo = "opcion_multiple"
            
    elif line.startswith("*(Texto:"):
        current_context = line.replace("*(Texto:", "").replace(")*", "").strip()
    elif line.startswith("*Texto:*"):
        current_context = line.replace("*Texto:*", "").replace('"', '').strip()
        
    elif line.startswith("**Pregunta"):
        # We found a question
        q_data = {
            "area": current_area,
            "tema": current_tema,
            "tipo": current_tipo,
            "pregunta": "",
            "opciones": [],
            "correcta": ""
        }
        if current_tema == "Comprensión Lectora" and current_context:
            q_data["contexto"] = current_context
            
        i += 1
        # Read question text
        q_text_lines = []
        while i < len(lines) and not lines[i].strip().startswith("a)") and not lines[i].strip().startswith("**Pregunta"):
            if lines[i].strip():
                q_text_lines.append(lines[i].strip())
            i += 1
            
        q_data["pregunta"] = " ".join(q_text_lines)
        
        # Read options
        opts = []
        while i < len(lines) and re.match(r'^[a-e]\)', lines[i].strip()):
            opt_text = re.sub(r'^[a-e]\)\s*', '', lines[i].strip())
            opts.append(opt_text)
            i += 1
            
        q_data["opciones"] = opts
        
        # Heurística para respuesta correcta (como no están marcadas en el .md, escogemos la 'a' temporalmente,
        # pero es peligroso. El usuario no puso respuestas correctas en el archivo MD, excepto donde hay una clave.)
        # En los exámenes, a veces la A no es la correcta.
        # Espera, si el MD no tiene las respuestas correctas indicadas, pondré la primera y luego
        # añadiremos un TODO para que se revisen.
        if len(opts) > 0:
            q_data["correcta"] = opts[0]
            questions.append(q_data)
        
        continue # skip the i += 1 at the end of loop
        
    i += 1

print(f"Parsed {len(questions)} questions.")

with open('temp_unsolved.json', 'w', encoding='utf-8') as f:
    json.dump(questions, f, indent=2, ensure_ascii=False)
    
print(f"Saved {len(questions)} parsed questions to temp_unsolved.json")

File: scripts/parsers/parse_all_admision.js
const fs = require('fs');

const content = fs.readFileSync('Raz verbal/preguntas_admision.md', 'utf8');
const lines = content.split('\n');

const questions = [];
let current_area = "Razonamiento Verbal";
let current_tema = "General";
let current_tipo = "opcion_multiple";
let current_context = "";

let i = 0;
while (i < lines.length) {
    const line = lines[i].trim();
    
    if (line.startsWith("### EXAMEN: DPTO. DE CIENCIAS")) {
        current_area = "Razonamiento Lógico";
        current_tema = "Resolución de Problemas";
    } else if (line.startsWith("#### SECCIÓN:")) {
        const tema_str = line.replace("#### SECCIÓN:", "").trim();
        if (tema_str.includes("ORTOGRAFÍA") || tema_str.includes("ACENTUACIÓN")) {
            current_tema = "Corrección Ortográfica";
            current_tipo = "comparacion_ortografica";
        } else if (tema_str.includes("SINÓNIMOS")) {
            current_tema = "Sinónimos en Contexto";
            current_tipo = "opcion_multiple";
        } else if (tema_str.includes("INCOMPLETAS") || tema_str.includes("COMPLETACIÓN")) {
            current_tema = "Completación de Oraciones";
            current_tipo = "opcion_multiple";
        } else if (tema_str.includes("ANALOGÍAS")) {
            current_tema = "Analogías y Relaciones Semánticas";
            current_tipo = "opcion_multiple";
        } else if (tema_str.includes("RAZONAMIENTO LÓGICO")) {
            current_tema = "Lógica Deductiva";
            current_tipo = "opcion_multiple";
        } else if (tema_str.includes("COMPRENSIÓN DE LECTURA")) {
            current_tema = "Comprensión Lectora";
            current_tipo = "opcion_multiple";
        } else if (tema_str.includes("ORDENACIÓN")) {
            current_tema = "Ordenación Lógica de Párrafos";
            current_tipo = "opcion_multiple";
        }
    } else if (line.startsWith("*(Texto:")) {
        current_context = line.replace("*(Texto:", "").replace(")*", "").trim();
    } else if (line.startsWith("*Texto:*")) {
        current_context = line.replace("*Texto:*", "").replace(/"/g, "").trim();
    } else if (line.startsWith("**Pregunta")) {
        const q_data = {
            area: current_area,
            tema: current_tema,
            tipo: current_tipo,
            pregunta: "",
            opciones: [],
            correcta: ""
        };
        
        if (current_tema === "Comprensión Lectora" && current_context) {
            q_data.contexto = current_context;
        }
        
        i++;
        const q_text_lines = [];
        while (i < lines.length && !lines[i].trim().match(/^[a-e]\)/) && !lines[i].trim().startsWith("**Pregunta")) {
            if (lines[i].trim()) {
                q_text_lines.push(lines[i].trim());
            }
            i++;
        }
        
        q_data.pregunta = q_text_lines.join(" ");
        
        const opts = [];
        while (i < lines.length && lines[i].trim().match(/^[a-e]\)/)) {
            const opt_text = lines[i].trim().replace(/^[a-e]\)\s*/, '');
            opts.push(opt_text);
            i++;
        }
        
        q_data.opciones = opts;
        if (opts.length > 0) {
            // Assigning temporary correct answer
            q_data.correcta = opts[0];
            questions.push(q_data);
        }
        continue;
    }
    i++;
}

console.log(`Parsed ${questions.length} questions.`);
fs.writeFileSync('temp_unsolved.json', JSON.stringify(questions, null, 2), 'utf8');
console.log(`Saved to temp_unsolved.json`);
